runelite-client/src/main/java/net/runelite/client/plugins/blastmine/BlastMineRockOverlay.java

```diff
@@ -68,28 +63,31 @@
 	private Color timerColor;
 	private Color warningColor;
 
-	@Inject
-	private ItemManager itemManager;
+	private final BufferedImage chiselIcon;
+	private final BufferedImage dynamiteIcon;
+	private final BufferedImage tinderboxIcon;
 
 	@Inject
-	BlastMineRockOverlay(@Nullable Client client, BlastMinePlugin plugin, BlastMinePluginConfig config)
+	BlastMineRockOverlay(Client client, BlastMinePlugin plugin, BlastMinePluginConfig config, ItemManager itemManager)
 	{
 		setPosition(OverlayPosition.DYNAMIC);
 		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.client = client;
 		this.plugin = plugin;
 		this.config = config;
-		this.client = client;
+		chiselIcon = itemManager.getImage(ItemID.CHISEL);
+		dynamiteIcon = itemManager.getImage(ItemID.DYNAMITE);
+		tinderboxIcon = itemManager.getImage(ItemID.TINDERBOX);
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		plugin.getRocks().removeIf(rock -> (rock.getRemainingTimeRelative() == 1 && rock.getType() != BlastMineRockType.NORMAL) || (rock.getRemainingFuseTimeRelative() == 1 && rock.getType() == BlastMineRockType.LIT));
 		drawRocks(graphics);
```

Comment:
Any reason for drawRocks to be split? as that is everything that this method contains, it should be here probably